data_structures/rb_tree2.cpp

```diff
@@ -0,0 +1,1490 @@
+/**
+ * @file rbtree2.cpp
+ * @author Brennan Allen (brennandoubt@gmail.com)
+ * @brief Program for a Red Black Tree algorithm, including a class with helper functions and separate test functions.
+ * @date 2022-08-26
+ * 
+ */
+
+#include <cassert>
+#include <random>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <sstream>
+#include <vector>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```